src/utilcode/util.cpp

```diff
@@ -1312,6 +1314,29 @@ DWORD_PTR GetCurrentProcessCpuMask()
 #endif
 }
 
+size_t GetOsPageSizeUncached()
+{
+    SYSTEM_INFO sysInfo;
+    ::GetSystemInfo(&sysInfo);
+    return sysInfo.dwAllocationGranularity ? sysInfo.dwAllocationGranularity : 0x1000;
+}
+
+size_t GetOsPageSize()
+{
+#ifdef FEATURE_PAL
+    // Initialization of function-local statics is guaranteed to occur only
+    // once even when called from multiple threads
+    //
+    // However there is a real potential performance cost.  This will need to
+    // be optimized if GetOsPageSize() shows up in profiles.
+    static const size_t pageSize(GetOsPageSizeUncached());
```

Comment:
@jkotas This is not intended to ignore your request.  Per @swgillespie 's suggestion I looked at the docs for std::call_once here. http://en.cppreference.com/w/cpp/thread/call_once.  It was noted that the standard makes
the guarantee noted in 1327-1328.  (Needs citation).  I recognize the potential performance impact you noted, but I do not know a good/simple way to guarantee this is initialized without using the function-local static guarantee.